File: android/app/src/main/kotlin/com/example/guidex/auth/AuthService.kt
package com.example.guidex.auth

import android.app.Activity
import android.content.Intent
import com.example.guidex.auth.model.RegisterStudentRequest
import com.example.guidex.auth.model.StudentProfile

interface AuthService {
    fun isLoggedIn(): Boolean
    fun getCurrentUid(): String?

    suspend fun registerWithEmail(request: RegisterStudentRequest): AuthResult<StudentProfile>
    suspend fun loginWithEmail(email: String, password: String): AuthResult<StudentProfile>
    suspend fun restoreSession(): AuthResult<StudentProfile?>

    fun googleSignInIntent(): Intent
    suspend fun signInWithGoogleResult(data: Intent?): AuthResult<StudentProfile>

    suspend fun signInWithApple(activity: Activity): AuthResult<StudentProfile>

    suspend fun logout()
}

File: lib/services/auth/auth_service.dart
import 'package:guidex/models/register_student_request.dart';
import 'package:guidex/models/student_profile.dart';

abstract class AuthService {
  Future<StudentProfile> registerWithEmail(RegisterStudentRequest request);

  Future<StudentProfile> signInWithEmail({
    required String email,
    required String password,
  });

  Future<StudentProfile> signInWithGoogle();

  Future<StudentProfile> signInWithApple();

  Future<StudentProfile?> restoreSession();

  Future<void> signOut();
}
